hbase_sql/src/main/java/com/splicemachine/derby/stream/spark/SparkHBaseBulkImport.java

```diff
@@ -0,0 +1,553 @@
+
+                List<Tuple2<Long, Tuple2<Double, ColumnStatisticsImpl>>> result = keyStatistics.collect();
+
+                // Calculate cut points for main table and index tables
+                cutPoints = getCutPoints(sampleFraction, result);
+
+                // dump cut points to file system for reference
+                dumpCutPoints(cutPoints);
+            }
+            if (!samplingOnly && !outputKeysOnly) {
+
+                // split table and indexes using the calculated cutpoints
+                if (cutPoints != null && cutPoints.size() > 0) {
+                    splitTables(cutPoints);
+                }
+
+
+                // get the actual start/end key for each partition after split
+                final List<BulkImportPartition> bulkImportPartitions =
+                        getBulkImportPartitions(allCongloms, bulkImportDirectory);
+
+                // Read data again and encode rows in main table and index
+                RowAndIndexGenerator rowAndIndexGenerator = new RowAndIndexGenerator(pkCols, tableVersion, execRow,
+                        autoIncrementRowLocationArray, spliceSequences, heapConglom, txn,
+                        operationContext, tentativeIndexList);
+                DataSet rowAndIndexes = dataSet.flatMap(rowAndIndexGenerator);
+                assert rowAndIndexes instanceof SparkDataSet;
+
+                // Create a data frame for main table and index key/values, and sort data
+                JavaRDD<Row> javaRdd =
+                        ((SparkDataSet)rowAndIndexes).rdd.map(
+                                new Function<Tuple2<Long,Tuple2<byte[], byte[]>>, Row>() {
+                                    @Override
+                                    public Row call(
+                                            Tuple2<Long,Tuple2<byte[], byte[]>> t) throws Exception {
+                                        
+                                        Long conglomerateId = t._1;
+                                        byte[] key = t._2._1;
+                                        byte[] value = t._2._2;
+                                        int partition = Collections.binarySearch(
+                                                bulkImportPartitions, new BulkImportPartition(conglomerateId, key, key, null),
+                                                BulkImportUtils.getSearchComparator());
+                                        return RowFactory.create(conglomerateId, Bytes.toHex(key), value, partition);
+                                    }
+                                });
+                SparkSession sparkSession = SparkSession.builder().getOrCreate();
+                SQLContext sqlContext = new SQLContext(sparkSession);
+                StructType schema = createSchema();
+                        Dataset<Row> rowAndIndexesDataFrame =
+                        sqlContext.createDataFrame(javaRdd, schema);
```

Comment:
No need to use a SQLContext, and you can get the SparkSession from the SpliceSpark class:
SpliceSpark.getSession().createDataFrame()